File: fixtures/js/empty_block/positive.ts
// Fixture: positive cases for MAINT013-empty-block (JS/TS)
// Each of the following should produce a finding.

const x = 1;

// Empty if body
if (x) {}

// Empty while body
while (false) {}

// Empty for body
for (let i = 0; i < 10; i++) {}

// Empty catch with named param (not intentional swallow)
function test() {
    try {
        doSomething();
    } catch (e) {}
}

declare function doSomething(): void;

File: fixtures/rust/empty_block/positive.rs
// Fixture: positive cases for MAINT013-empty-block (Rust)
// Each of the following should produce a finding.

fn check_empty_if(x: bool) {
    if x {}
}

fn check_empty_for() {
    for _i in 0..10 {}
}

fn check_empty_while(mut x: i32) {
    while x > 0 {}
}
